commands/issue.go

```diff
@@ -48,80 +56,146 @@ var (
 }
 
-/*
-  $ hub issue
-*/
-func issue(cmd *Command, args *Args) {
-	runInLocalRepo(func(localRepo *github.GitHubRepo, project *github.Project, gh *github.Client) {
-		if args.Noop {
-			ui.Printf("Would request list of issues for %s\n", project)
-		} else {
-			issues, err := gh.Issues(project)
-			utils.Check(err)
-			for _, issue := range issues {
-				var url string
-				// use the pull request URL if we have one
-				if issue.PullRequest.HTMLURL != "" {
-					url = issue.PullRequest.HTMLURL
-				} else {
-					url = issue.HTMLURL
-				}
+func listIssues(cmd *Command, args *Args) {
+	localRepo, err := github.LocalRepo()
+	utils.Check(err)
 
-				if flagIssueAssignee == "" ||
-					strings.EqualFold(issue.Assignee.Login, flagIssueAssignee) {
-					// "nobody" should have more than 1 million github issues
-					ui.Printf("% 7d] %s ( %s )\n", issue.Number, issue.Title, url)
-				}
+	project, err := localRepo.MainProject()
+	utils.Check(err)
+
+	gh := github.NewClient(project.Host)
+
+	if args.Noop {
+		ui.Printf("Would request list of issues for %s\n", project)
+	} else {
+		filters := map[string]interface{}{}
+		if cmd.FlagPassed("state") {
+			filters["state"] = flagIssueState
+		}
+		if cmd.FlagPassed("assignee") {
+			filters["assignee"] = flagIssueAssignee
+		}
+
+		issues, err := gh.FetchIssues(project, filters)
+		utils.Check(err)
+
+		maxNumWidth := 0
+		for _, issue := range issues {
+			if numWidth := len(strconv.Itoa(issue.Number)); numWidth > maxNumWidth {
+				maxNumWidth = numWidth
 			}
 		}
-	})
-}
 
-func createIssue(cmd *Command, args *Args) {
-	runInLocalRepo(func(localRepo *github.GitHubRepo, project *github.Project, gh *github.Client) {
-		if args.Noop {
-			ui.Printf("Would create an issue for %s\n", project)
-		} else {
-			title, body, err := getTitleAndBodyFromFlags(flagIssueMessage, flagIssueFile)
-			utils.Check(err)
-
-			if title == "" {
-				title, body, err = writeIssueTitleAndBody(project)
-				utils.Check(err)
+		colorize := ui.IsTerminal(os.Stdout)
+
+		for _, issue := range issues {
+			if issue.PullRequest != nil {
+				continue
+			}
+
+			num := fmt.Sprintf("#%d", issue.Number)
+			numWidth := len(num)
+
+			if colorize {
+				issueColor := 32
+				if issue.State == "closed" {
+					issueColor = 31
+				}
+				num = fmt.Sprintf("\033[%dm%s\033[0m", issueColor, num)
 			}
 
-			issue, err := gh.CreateIssue(project, title, body, flagIssueLabels)
-			utils.Check(err)
+			ui.Printf("%*s%s  %s", maxNumWidth+1-numWidth, "", num, issue.Title)
+
+			for i, label := range issue.Labels {
+				color, err := utils.NewColor(label.Color)
+				if err != nil {
+					utils.Check(err)
+				}
+
+				textColor := 16
+				if color.Brightness() < 0.65 {
+					textColor = 15
+				}
+
+				if i == 0 {
+					ui.Printf(" ")
+				}
+				ui.Printf(" \033[38;5;%d;48;2;%d;%d;%dm %s \033[m", textColor, color.Red, color.Green, color.Blue, label.Name)
+			}
 
-			ui.Println(issue.HTMLURL)
+			ui.Printf("\n")
 		}
-	})
+	}
+
+	os.Exit(0)
```

Comment:
You're right. This isn't a huge code debt yet (since it's just a few identical lines like this throughout the codebase), but we should definitely solve this through CmdRunner system going forward.